llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -16266,7 +16266,8 @@ SDValue DAGCombiner::visitFREEZE(SDNode *N) {
   // Finally, recreate the node, it's operands were updated to use
   // frozen operands, so we just need to use it's "original" operands.
   SmallVector<SDValue> Ops(N0->ops());
-  // Special-handle ISD::UNDEF, each single one of them can be it's own thing.
+  // Special-handle ISD::UNDEF, ISD::POISON, each single one of them can be it's
```

Comment:
This comment is kind of wrong. Undef and poison should get separate handling, but best leave for a future patch. Add a todo?

Reply:
I missed that this is in the context of the freeze combine, so the comment doesn't really apply.